Sources/MapboxCoreNavigation/NavigationService.swift

```diff
@@ -382,18 +336,75 @@ public class MapboxNavigationService: NSObject, NavigationService {
         eventsManager.activeNavigationDataSource = self
         eventsManager.resetSession()
     }
-
-    private func resetGPSCountdown() {
-        //Sanity check: if we're not on this mode, we have no business here.
-        guard simulationMode == .onPoorGPS else { return }
+    
+    /**
+     Intializes a new `NavigationService`. Useful convienence initalizer for OBJ-C users, for when you just want to set up a service without customizing anything.
+     
+     - parameter routeResponse: `RouteResponse` object, containing selection of routes to follow.
+     - parameter routeIndex: The index of the route within the original `RouteResponse` object.
+     - parameter routeOptions: The route options used to get the route.
+     */
+    convenience init(routeResponse: RouteResponse, routeIndex: Int, routeOptions options: RouteOptions) {
+        self.init(routeResponse: routeResponse, routeIndex: routeIndex, routeOptions: options, directions: nil, locationSource: nil, eventsManagerType: nil)
+    }
+    
+    // MARK: Serving the Lifecycle
```

Comment:
This isn’t very intuitive, but it seems like these methods can all go under “Starting and Stopping Navigation” anyways.